pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -277,3 +294,44 @@ func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentCo
 	glog.V(4).Infof("Updated the status for %q (observed generation: %d)", deployutil.LabelForDeploymentConfig(config), config.Status.ObservedGeneration)
 	return nil
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeploymentList *kapi.ReplicationControllerList, deploymentConfig *deployapi.DeploymentConfig) error {
+	activeDeployment := deployutil.ActiveDeployment(deploymentConfig, existingDeployments)
+
+	// we need to coerce the list type into something we can sort, and we need to filter out
+	// any deployments that we don't care about, namely the active deployment and any newer
+	// deployments
+	existingDeployments := []*kapi.ReplicationController{}
```

Comment:
existingDeploymentList contains the active deployment, existingDeployments (which should be oldDeployments) contains all candidates for cleaning up.